process/esdtSupplyProcessor.go

```diff
@@ -75,12 +77,33 @@ func (esp *esdtSupplyProcessor) getSupplyFromShards(tokenIdentifier string) (*bi
 			return nil, err
 		}
 
-		totalSupply.Add(totalSupply, supply)
+		addToSupply(totalSupply, supply)
 	}
 
 	return totalSupply, nil
 }
 
+func addToSupply(dstSupply, sourceSupply *data.ESDTSupply) {
+	dstSupply.Supply = sumStr(dstSupply.Supply, sourceSupply.Supply)
+	dstSupply.Burned = sumStr(dstSupply.Burned, sourceSupply.Burned)
+	dstSupply.Minted = sumStr(dstSupply.Minted, sourceSupply.Minted)
+
+	return
+}
+
+func sumStr(s1, s2 string) string {
+	s1Big, ok := big.NewInt(0).SetString(s1, 10)
+	if !ok {
+		return s2
```

Comment:
is it ok to return these here? Why not signal through error that something went wrong and return 0?

Reply:
I think it's ok this way:
"NaN" + "10" = "10"
"10" + "NaN" = "10"

but a log warn would be helpful